linalg/vector.hpp

```diff
@@ -67,7 +67,7 @@ class Vector
 public:
 
    /// Default constructor for Vector. Sets size = 0 and data = NULL.
-   Vector() { data.Reset(); size = 0; }
+   Vector(): size(0) { }
```

Comment:
Maybe add a comment here that the default constructor for `data` will zero-out all the `Memory` metadata and host pointer? (It says it sets `data = NULL`, but that happens only implicitly).

Reply:
Tell me if you like it more.

Reply:
Looks great